custom_components/sector/event.py

```diff
@@ -17,23 +19,43 @@ async def async_setup_entry(
     hass: HomeAssistant,
     entry: SectorAlarmConfigEntry,
     async_add_entities: AddEntitiesCallback,
-) -> None:
-    """Set up Sector Alarm event entities."""
+):
+    """Set up a single event entity per device in Sector Alarm coordinator."""
     coordinator: SectorDataUpdateCoordinator = entry.runtime_data
-    devices = coordinator.data.get("devices", {})
-    logs = coordinator.data.get("logs", {})
+    grouped_events = await coordinator.process_events()
     entities = []
 
-    # Create event entities for each supported device with event logs
-    for serial_no, device_info in devices.items():
-        if device_info.get("model") == "Smart Lock":  # Filter for Smart Locks
-            # Check if there are logs for this Smart Lock to create event entities
-            if serial_no in logs:
-                entities.append(SectorAlarmEvent(coordinator, serial_no, device_info))
-                _LOGGER.debug(
-                    "SECTOR_EVENT: Created event entity for Smart Lock with serial: %s",
-                    serial_no,
-                )
+    for device_serial, event_categories in grouped_events.items():
+        device_info = coordinator.get_device_info(device_serial)
+        device_name = device_info["name"]
+        device_model = device_info["model"]
+
+        _LOGGER.debug(
+            "SECTOR_EVENT: Creating event entity for device: %s, Model: %s",
+            device_name,
+            device_model,
+        )
+
+        # Define unique entity ID for each device, regardless of event type
+        entity_unique_id = f"{device_serial}_event"
+
+        # Check if the entity already exists by unique ID
+        if hass.states.get(f"event.{entity_unique_id}"):
+            _LOGGER.debug(
+                "SECTOR_EVENT: Entity with unique ID '%s' already exists, skipping.",
+                entity_unique_id,
+            )
+            continue
```

Comment:
I think we should just replace here to add an event entity for each lock and then it's done.